plugins/edsm.py

```diff
@@ -12,135 +12,191 @@
 
-def plugin_app(parent):
+
+def plugin_app(parent: tk.Tk) -> None:
     this.system_link = parent.children['system']  # system label in main window
     this.system_link.bind_all('<<EDSMStatus>>', update_status)
     this.station_link = parent.children['station']  # station label in main window
 
-def plugin_stop():
+
+def plugin_stop() -> None:
     # Signal thread to close and wait for it
     this.queue.put(None)
     this.thread.join()
     this.thread = None
     # Suppress 'Exception ignored in: <function Image.__del__ at ...>' errors
     this._IMG_KNOWN = this._IMG_UNKNOWN = this._IMG_NEW = this._IMG_ERROR = None
 
-def plugin_prefs(parent, cmdr, is_beta):
 
-    PADX = 10
-    BUTTONX = 12	# indent Checkbuttons and Radiobuttons
-    PADY = 2		# close spacing
+def plugin_prefs(parent: tk.Tk, cmdr: str, is_beta: bool) -> tk.Frame:
+    PADX = 10  # noqa: N806
+    BUTTONX = 12  # indent Checkbuttons and Radiobuttons # noqa: N806
+    PADY = 2		# close spacing # noqa: N806
 
     frame = nb.Frame(parent)
     frame.columnconfigure(1, weight=1)
 
-    HyperlinkLabel(frame, text='Elite Dangerous Star Map', background=nb.Label().cget('background'), url='https://www.edsm.net/', underline=True).grid(columnspan=2, padx=PADX, sticky=tk.W)	# Don't translate
-    this.log = tk.IntVar(value = config.getint('edsm_out') and 1)
-    this.log_button = nb.Checkbutton(frame, text=_('Send flight log and Cmdr status to EDSM'), variable=this.log, command=prefsvarchanged)
-    this.log_button.grid(columnspan=2, padx=BUTTONX, pady=(5,0), sticky=tk.W)
+    HyperlinkLabel(
+        frame,
+        text='Elite Dangerous Star Map',
+        background=nb.Label().cget('background'),
+        url='https://www.edsm.net/',
+        underline=True
+    ).grid(columnspan=2, padx=PADX, sticky=tk.W)  # Don't translate
+
+    this.log = tk.IntVar(value=config.getint('edsm_out') and 1)
+    this.log_button = nb.Checkbutton(
+        frame, text=_('Send flight log and Cmdr status to EDSM'), variable=this.log, command=prefsvarchanged
+    )
+
+    this.log_button.grid(columnspan=2, padx=BUTTONX, pady=(5, 0), sticky=tk.W)
+
+    nb.Label(frame).grid(sticky=tk.W)  # big spacer
+    # Section heading in settings
+    this.label = HyperlinkLabel(
+        frame,
+        text=_('Elite Dangerous Star Map credentials'),
+        background=nb.Label().cget('background'),
+        url='https://www.edsm.net/settings/api',
+        underline=True
+    )
 
-    nb.Label(frame).grid(sticky=tk.W)	# big spacer
-    this.label = HyperlinkLabel(frame, text=_('Elite Dangerous Star Map credentials'), background=nb.Label().cget('background'), url='https://www.edsm.net/settings/api', underline=True)	# Section heading in settings
     this.label.grid(columnspan=2, padx=PADX, sticky=tk.W)
 
-    this.cmdr_label = nb.Label(frame, text=_('Cmdr'))	# Main window
+    this.cmdr_label = nb.Label(frame, text=_('Cmdr'))  # Main window
     this.cmdr_label.grid(row=10, padx=PADX, sticky=tk.W)
```

Comment:
Rather than magic row numbers change the code to have a row variable incremented after each row is added.  It avoids issues with forgetting to change row numbers if you insert or delete something.